d2l/train.py

```diff
@@ -47,26 +46,30 @@ def train_ch3(net, train_iter, test_iter, criterion, num_epochs, batch_size, lr=
 
 
 def train_ch5(net, train_iter, test_iter, criterion, num_epochs, batch_size, device, lr=None):
-    net = net.to(device)
+    """Train and evaluate a model with CPU or GPU."""
     print('training on', device)
+    net.to(device)
     optimizer = optim.SGD(net.parameters(), lr=lr)
     for epoch in range(num_epochs):
-        net.train() # Switch to training mode
-        train_l_sum, train_acc_sum, n, start = 0.0, 0.0, 0, time.time()
+        train_l_sum = torch.tensor([0.0],dtype=torch.float32,device=device)
+        train_acc_sum = torch.tensor([0.0],dtype=torch.float32,device=device)
+        n, start = 0, time.time()
         for X, y in train_iter:
-            X, y = X.to(device), y.to(device)
+            net.train() # Switch to training mode
```

Comment:
Again like net.eval please take it to the original position. We don't want to call this in each loop for a single iterator. Take it below the line 
```
for epoch in range(num_epochs):
    net.train() # Switch to training mode

```

Reply:
The following Error is resolved.